src/ray/common/cgroup2/cgroup_driver_interface.h

```diff
@@ -0,0 +1,197 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "ray/common/status.h"
+#include "ray/common/status_or.h"
+
+namespace ray {
+
+/**
+    A utility interface that allows the caller to perform cgroup operations.
+    TODO(irabbani): I need to figure out the following
+        1. What is the appropriate level of documentation here? This header file is what's
+        important by consumers of the API and it needs to provide the correct docstrings
+        and API.
+        2. Revisiting this API to make it sure it integrates with a possible systemd
+        driver implementation (meaning it cannot have any directory based info or any dbus
+        implementation).
+ */
+class CgroupDriverInterface {
+ public:
+  virtual ~CgroupDriverInterface() = default;
+
+  /**
+    Checks to see if only cgroupv2 is enabled (known as unified mode) on the system.
+    If cgroupv2 is not enabled, or enabled along with cgroupv1, returns Invalid
+    with the appropriate error message.
+
+    @see systemd's documentation for more information about unified mode:
+    https://github.com/systemd/systemd/blob/main/docs/CGROUP_DELEGATION.md#hierarchy-and-controller-support
+
+    @see K8S documentation on how to enable cgroupv2 and check if it's enabled correctly:
+    https://kubernetes.io/docs/concepts/architecture/cgroups/#linux-distribution-cgroup-v2-support
+
+    @return OK if successful, otherwise Invalid.
+    */
+  virtual Status CheckCgroupv2Enabled() = 0;
+
+  /**
+    Checks that the cgroup is valid. See return values for details of which
+    invariants are checked.
+
+    @param cgroup the absolute path of the cgroup.
+
+    @return OK if no errors are encounted. Otherwise, one of the following errors
+    NotFound if the cgroup does not exist.
+    PermissionDenied if current user doesn't have read, write, and execute permissions.
+    InvalidArgument if the cgroup is not using cgroupv2.
+   */
+  virtual Status CheckCgroup(const std::string &cgroup) = 0;
```

Comment:
Is there any need to get the metadata of a specific cgroup? If so, I would generalize this to `GetCgroup` and return the status + metadata.